### PR TITLE
Allow strong params in ActiveRecord::Base#exists?

### PR DESCRIPTION
**Pull request by gmcgibbon**
### Summary

Fixes https://github.com/rails/rails/issues/34752.

Allows `ActionController::Params` as argument of `ActiveRecord::Base#exists?`. This works already for `#create` and `#where`, so I think this change makes sense.

### CODE REVIEW

File: activerecord/test/cases/finder_test.rb [hunk 1]
```diff
@@ -224,6 +225,14 @@ def test_exists_with_string
     assert_equal true, Subscriber.exists?("   ")
   end
 
+  def test_exists_with_strong_parameters
+    assert_equal false, Subscriber.exists?(nick: "foo")
```
Comment (rafaelfranca) [inline activerecord/test/cases/finder_test.rb]: Am I missing something or this test is not using `Parameters`?

Comment (gmcgibbon) [reply activerecord/test/cases/finder_test.rb]: 🤦‍♂️ My bad, fixed now.

**Comment by kaspth** [comment]
I thought we had deprecated exists? without arguments?

**Comment by rafaelfranca** [comment]
> I thought we had deprecated exists? without arguments?

Not that I'm aware. Why would we do that?

**Comment by kaspth** [comment]
I can't for the life of me find that discussion now via GitHub's search. I just seem to remember @matthewd and either @eugeneius and @kamipo talking about this.

**Pull request merged**